### PR TITLE
feat: arb1 uniswap/coingecko mapped lists

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

- Refactored the script that was used to map Omnibridge from `mainnet` to `Gnosis chain` to also map `mainnet` to `Arbitrum one`
- Ran the scripts and added the resulting token lists
- Updated Coingecko.json
- Updated all Permit info after with the bridge lists

# To test

1. Get the raw github file url for the new token lists
2. Add it to cowswap
* List should be valid
* All tokens should be tradable, although I don't expect anyone to check the 100+ tokens.

### CODE REVIEW

File: src/scripts/generateBridgeList.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    ARBITRUM_BRIDGE_ABI,
+    'calculateL2TokenAddress',
+    resultFile,
+    {
+      // USDC
+      '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48': '0xaf88d065e77c8cc2239327c5edb3a432268e5831',
+      // WETH
+      '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2': '0x82af49447d8a07e3bd95bd0d56f35241523fbab1',
+    },
+  )
```
Comment (alfetopito) [inline src/scripts/generateBridgeList.ts:159]: Manual overwrites, for known tokens for which we shouldn't use the bridged info.

File: src/scripts/generateBridgeList.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        // Copy everything from the Mainnet token and override address and chainId
+        // Let's hope no one deploys the same token in a different networks with different data
+
+        const extensions = token.extensions as Extensions
+        const tokenAddress = extensions?.bridgeInfo?.[chainId]?.tokenAddress
+
+        if (tokenAddress && tokenAddress.toLowerCase() !== address.toLowerCase()) {
+          console.log(
+            `TokenList extension contains address different than returned by bridge. Extension: ${tokenAddress} Bridge: ${address}`,
+          )
```
Comment (alfetopito) [inline src/scripts/generateBridgeList.ts:101]: Uniswap token list includes the address of bridged tokens for some chains.
Use that when available, if different from the bridged address.

File: src/scripts/generateBridgeList.ts [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            return acc
+          }
+
+          // Replace the fetched address with the one provided
+          acc.push({
+            ...token,
+            chainId,
+            address: toReplace,
+            extensions: undefined,
+          })
```
Comment (alfetopito) [inline src/scripts/generateBridgeList.ts:88]: Apply the manual address overwrite

File: src/abi/abitrumBridgeAbi.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export const ARBITRUM_BRIDGE_ABI = [
+  {
+    inputs: [{ internalType: 'address', name: 'l1ERC20', type: 'address' }],
+    name: 'calculateL2TokenAddress',
```
Comment (alfetopito) [inline src/abi/abitrumBridgeAbi.ts:4]: Different than Omnibridge contract, this one always return an address. We might add tokens which have not been bridged yet.

File: src/scripts/generateBridgeList.ts (hunk 1 above)
Comment (shoom3301) [reply src/scripts/generateBridgeList.ts:159]: Would be great to move them into some json file

Comment (shoom3301) [reply src/scripts/generateBridgeList.ts:159]: Why only USDC and WETH?

Comment (alfetopito) [reply src/scripts/generateBridgeList.ts:159]: Those have their native versions on arb1

- WETH is the wrapped native ETH
- USDC has a native deployment as well

There might be others.
Actually, there are others
image

Should I hard code them all?

Comment (shoom3301) [reply src/scripts/generateBridgeList.ts:159]: Wow, I didn't know. I'm ok with the current state

Comment (alfetopito) [reply src/scripts/generateBridgeList.ts]: I added them all

File: src/scripts/generateBridgeList.ts [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+  generateBridgedList(
+    SupportedChainId.ARBITRUM_ONE,
+    source,
+    ARBITRUM_BRIDGE_ADDRESS,
+    ARBITRUM_BRIDGE_ABI,
+    'calculateL2TokenAddress',
+    resultFile,
+    {
+      // USDC
```
Comment (shoom3301) [inline src/scripts/generateBridgeList.ts:154]: Those mappings could be extracted to some json file

**Review by shoom3301 — APPROVED** [review]

**Pull request merged**